src/main/java/jenkins/branch/MultiBranchProject.java

```diff
@@ -538,9 +553,17 @@ public P getItem(String name) {
             if (item != null) {
                 return item;
             }
+            item = super.getItem(NameMangler.apply(decoded));
```

Comment:
Does not seem to make sense. Should this not be un-mangling, and doing so on `name` rather than `decoded`? Hard to follow what all this logic really is doing; does it have full test coverage?